### ISSUE TITLE
Roadmap

### ISSUE THREAD

**Issue by xavdid** [vdavez/codingforlawyers#23]
Love this! 

I'm definitely interested in contributing some chapters (I love the idea of teaching people coding and have been way behind on updating my blog posts on the subject (http://blog.davidbrownman.com/post/64233112932/what-the-hell-is-a-python)). 

Going forward, it seems like you (as our fearless leader) should provide a few resources (or we should discuss what we want to do):
- An overall goal/endpoint
- Table of contents so we can begin to move towards that goal
- Keeping issues maintained (I noticed there are a few issues that are resolved but not closed- a clean repo is a happy repo!)

This seems like a great project though and I'm excited to dive into it.

**Comment by adelevie** [comment]
Perhaps it could be advertised better, but there's a "planned table of contents" over in the [wiki](https://github.com/vzvenyach/codingforlawyers/wiki/Table-of-Contents).

**Comment by eah13** [comment]
Agreed!  Guidelines on content would be helpful too.  I love the interactive examples in Chapter 4 and would be happy to make other sections' examples interactive if that's helpful..  My interactive version of Google's Python course has some good regex material:

https://elliott.trinket.io/google-s-python-class#/regular-expressions/python-regular-expressions

Also @trinketapp could probably build a way to make code blocks interactive without needing the iframes like we did with [Jekyll tools](https://github.com/trinketapp/jekyll-tools)

**Comment by xavdid** [comment]
@adelevie That's good that it's there, but it's also not any farther than we are right now. 

That is, if I wanted to write a chapter tonight, I wouldn't know what to write. I propose we lay out the whole ToC now and then fill in and edit as needed.

**Comment by vdavez** [comment]
First off, thanks for the issue and for your enthusiasm. I'm pysched to have you contribute! 

I have updated the wiki, https://github.com/vzvenyach/codingforlawyers/wiki/Table-of-Contents, and encourage others to feel free to comment on the TOC or open new issues.

I've also tried to close out the old issues and PRs. I'm a bit limited in my ability to work on things before the evening, but I'll try and get to them quickly.

Part of the project is that it's evolved. I have a very specific sense of _how_ to present material, and I've thought a lot about pace/content. But I am very open to suggestions about ways to reorganize/make it more pedagogically sound.

Thanks again!

**Comment by xavdid** [comment]
Awesome! And I didn't mean to rag on you about the PR's- I forget sometimes that not everyone sits on GH all day as part of their job, haha. 

In terms of organization, I would think we would drive people towards a goal- probably deploying a web app? Using heroku, we would incorporate git (heroku deploys are git repos), html/css, some sort of server, and actually getting running code out into the world. 

In the short term, we could put a "contributing" section in the readme that points users towards the ToC, and any open issues as places to start.

**Comment by eah13** [comment]
@vzvenyach your vision for the organization is what it's all about.  Maybe there could be a 'Contributed Articles' section that ws more liberal, and successful ones you like could get merged into the main part of the site if/when you feel they're ready.  This is similar to how programming languages often have a 'contrib' library that has a lower bar or sometimes wider goals than the main project.

Anyways, whatever you choose, clear guidelines for contributing will help the enthusiasm push the project the direction you and the other committers want for it.

**Comment by compleatang** [comment]
Amazing to wake up to all this discussion! From my perspective at least, while it is always neat to deploy web apps, part of what is great about what @vzvenyach is trying to do here is to show lawyers that programming is actually relevant to their practice. I know for a fact that programming small tools has made me a more efficient (and I hope, effective) lawyer. 

Part of that is mindset -- DRY, templating, etc. have made me fundamentally rethink how I practice transactional law. Part of that is my ability to drop into a scripting language and crank out a script to do something for me. 

If we are dropping wishes into the roadmap bucket mine is that lawyers finish reading this site|book with the sense that they can (a) build a thing or two and (b) that programming will help them be more efficient lawyers.
